subtests/docker_cli/build/build.py

```diff
@@ -114,213 +124,191 @@ def dockerfile_repo_replace(self, path, repo):
+                         use_config_repo=use_config_repo,
+                         dockerfile_dir_path=dockerfile_dir_path,
+                         base_repo_fqin=base_repo_fqin,
+                         positive_build_regex=positive_build_regex,
+                         negative_build_regex=negative_build_regex)]
+
+    def initialize(self):
+        super(build_base, self).initialize()
+        self.initialize_utils()
+        self.sub_stuff['builds'] = []
+        self.sub_stuff['builds'] += self.make_builds(self.config)
 
     def run_once(self):
         super(build_base, self).run_once()
-        # Run single build
-        self._build_container(self.sub_stuff['builds'][0],
-                              [self.config['docker_build_options']])
-
-    def _build_container(self, build_def, subargs):
-        """
-        Build container according to the `build_def` dictionary.
-        """
-        subargs += ["-t", build_def['image_name'],
-                    build_def['dockerfile_dir_path']]
-        dkrcmd = DockerCmd(self, 'build', subargs,
-                           self.config['build_timeout_seconds'],
-                           verbose=True)
-        build_def['result'] = dkrcmd.execute()
-
-    def _postprocess_exit_status(self, build_def):
-        """
-        Check the exit status and eventually the build_fail_msg.
-        """
-        if build_def.get('build_fail_msg'):
-            self.failif(build_def['result'].exit_status == 0, "Build returned "
-                        "0 even thought it was expected to fail: %s"
-                        % build_def['result'])
-            out = "%s\n%s" % (build_def['result'].stdout,
-                              build_def['result'].stderr)
-            exp = build_def.get('build_fail_msg')
-            self.failif(exp not in out, "Expected failure message '%s' not "
-                        "found in the build output:\n%s"
-                        % (exp, build_def['result']))
-        else:
-            OutputGood(build_def['result'])
-            self.failif(build_def['result'].exit_status != 0, "Non-zero build "
-                        "exit status: %s" % build_def['result'])
-        self.logdebug("%s:\tExit status\tOK", build_def['image_name'])
-
-    def _postprocess_created_images(self, build_def):
-        """
-        Check that all expected images were created
-        """
+        for build_def in self.sub_stuff['builds']:
+            build_def.dockercmd.execute()
+
+    def postprocess_positive(self, build_def):
+        # Verify exit status
+        mustpass(build_def.dockercmd.cmdresult)
+        OutputGood(build_def.dockercmd.cmdresult)
+        # Fail if does not match positive regex
+        if build_def.positive_build_regex != '':
+            positive_regex = re.compile(build_def.positive_build_regex)
+            for output in (build_def.dockercmd.stdout,
+                           build_def.dockercmd.stderr):
```

Comment:
...okay, fixed this particular method, so it takes a parameter to help decide on checking stdout or stderr...